### DISCUSSION TITLE
Can launch command support Dialog/Toast windows?

### DISCUSSION THREAD

**Discussion by salmankhilji** [kovidgoyal/kitty#5944] [Ideas]
How difficult would it be for kitty to support a new dialog/toast window type in the `launch` command?

These will be modal windows closely resembling windows of type `overlay` or `overlay-main` with user-defined margins on all 4 sides.

We can launch a new overlay  window with margins as `top=bottom=height/4` and `left=right=width/4` to create a dialog box centered on the active window.  Or perhaps with margins as `top=height*3/4` and `left,=right=bottom=0` to produce an Android toast like window at the bottom.

Primary use-case is user input, especially with the `ask` kitten.  Working with kitty OS windows started mostly as full-screen, I personally find the `create_marker` overlay windows unnecessarily large on my 27" monitor.

If it is something you would like to do, here are a few items that would be icing on the cake:

1. Ability to darken the main window which has a non-full screen dialog/toast on top
2. Ability to specify slide in/out transitions
3. Ability to specify corner radius

**Comment by kovidgoyal** [comment]
Not something I care about. Patches welcome.
